rules/aws_s3_bucket_name.go

```diff
@@ -81,7 +81,7 @@ func (r *AwsS3BucketNameRule) Check(runner tflint.Runner) error {
 	}{
 		{
 			Regexp: *regexp.MustCompile(fmt.Sprintf(
-				"^(.{0,%d})?(.{%d,})?$",
+				"^.{0,%d}$|.{%d,}$",
```

Comment:
Found out about the alternation operator in Go and thought it was easier to read and more go-friendly.